src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,298 @@
+                fileWriter.write(currentTask.getType() + " | " + ((currentTask.isDone()) ? "1" : "0") + " | "
+                        + currentTask.getDescription() + ((currentTask.hasDate()) ? " | " + currentTask.getDate() : "") + '\n' );
+            }
+            fileWriter.close();
+            System.out.println("\tData saved");
+            return true;
+        } catch (IOException e) {
+            System.out.println("\tAn error occurred while saving your data :(");
+            return false;
+        }
+    }
+
+    public static void takeInput() {
+        Scanner userInput = new Scanner(System.in);
+        while (userInput.hasNextLine()) {
+            String command = userInput.nextLine();
+            if (command.equals("bye") || command.equals("exit")) {
+                dividerLine();
+                if (exitDuke()) {
+                    dividerLine();
+                    userInput.close();
+                    break;
+                }
+                dividerLine();
+            } else if (command.equals("list")) {
+                dividerLine();
+                list();
+                dividerLine();
+            } else if (command.startsWith("done")) {
+                dividerLine();
+                finishTask(command);
+                dividerLine();
+            } else if (command.startsWith("todo")) {
+                dividerLine();
+                addTodo(command);
+                dividerLine();
+            } else if (command.startsWith("event")) {
+                dividerLine();
+                addEvent(command);
+                dividerLine();
+            } else if (command.startsWith("deadline")) {
+                dividerLine();
+                addDeadline(command);
+                dividerLine();
+            } else if (command.startsWith("delete")) {
+                dividerLine();
+                deleteTask(command);
+                dividerLine();
+            } else if (command.equals("save")) {
+                dividerLine();
+                saveData();
+                dividerLine();
+            } else {
+                dividerLine();
+                unknownCommand();
+                dividerLine();
+            }
+        }
+    }
+    
+    public static String printUsage(String type) {
+        switch(type) {
+        case "todo":
+            return "todo [task description]";
+        case "event":
+            return "event [event description] /at [date]";
+        case "deadline":
+            return "deadline [task description] /by [due date]";
+        case "done":
+            return "done [index of task]";
+        case "delete":
+            return "delete [index of task]";
+        default:
+            return null;
+        }
+    }
+    
+    public static void unknownCommand() {
+        System.out.println("\tSorry, this command is not available yet. Stay tuned! :)");
+    }
+    
+    public static void list() {
+        if (tasks.size()==0) {
```

Comment:
Consider making the happy path prominent in this code by checking whether there are any tasks in the list and returning to the call stack if there aren't any. Then you may eliminate the need for the else case clause and reduce a level of indentation. (There are multiple instances of this in the code so you similarly do this for those instances too)

https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent